lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py

```diff
@@ -417,11 +549,26 @@ def apply(self, resource_reqs: ResourceRequirements) -> None:
             field_manager=self.__class__.__name__,
         )
 
+    def dry_run_apply(self, resource_reqs: ResourceRequirements):
+        """Run a dry-run patch operation."""
+        self.client.patch(
+            StatefulSet,
+            self.statefulset_name,
+            self._patched_delta(resource_reqs),
+            namespace=self.namespace,
+            patch_type=PatchType.APPLY,
+            field_manager=self.__class__.__name__,
+            dry_run=True,
+        )
```

Comment:
Is there a way to consolidate this code with the real apply?
To avoid diverging in case we forget to update something in _both_ places.